packages/thirdweb/src/extensions/marketplace/write/direct/updateListing.ts

```diff
@@ -0,0 +1,196 @@
+import type { Address } from "abitype";
+import {
+  NATIVE_TOKEN_ADDRESS,
+  isNativeTokenAddress,
+} from "../../../../constants/addresses.js";
+import { getContract } from "../../../../contract/contract.js";
+import { eth_getBlockByNumber } from "../../../../rpc/actions/eth_getBlockByNumber.js";
+import { getRpcClient } from "../../../../rpc/rpc.js";
+import type { BaseTransactionOptions } from "../../../../transaction/types.js";
+import { toUnits } from "../../../../utils/units.js";
+import { isERC721 } from "../../../erc721/read/isERC721.js";
+import { isERC1155 } from "../../../erc1155/read/isERC1155.js";
+import { updateListing as generatedUpdateListing } from "../../__generated__/IDirectListings/write/updateListing.js";
+import { getListing } from "../../read/direct/getListing.js";
+
+export type UpdateListingParams = {
+  /**
+   * The ID of the listing to update
+   */
+  listingId: bigint;
+  /**
+   * The contract address of the asset being listed
+   */
+  assetContractAddress?: Address;
+  /**
+   * The ID of the token being listed
+   */
+  tokenId?: bigint;
+  /**
+   * The quantity of tokens to list
+   *
+   * For ERC721s, this value can be omitted.
+   * @default 1
+   */
+  quantity?: bigint;
+  /**
+   * The contract address of the currency to accept for the listing
+   * @default NATIVE_TOKEN_ADDRESS
+   */
+  currencyContractAddress?: Address;
+  /**
+   * The start time of the listing
+   * @default new Date()
+   */
+  startTimestamp?: Date;
+  /**
+   * The end time of the listing
+   * @default new Date() + 10 years
+   */
+  endTimestamp?: Date;
+  /**
+   * Whether the listing is reserved to be bought from a specific set of buyers
+   * @default false
+   */
+  isReservedListing?: boolean;
+} & (
+  | {
+      /**
+       * The price per token (in Ether)
+       */
+      pricePerToken: string;
+    }
+  | {
+      /**
+       * The price per token (in wei)
+       */
+      pricePerTokenWei: string;
```

Comment:
really need to standardize on this everywhere, will do it for claim conditions stuff